packages/ui/src/components/Tooltip/index.tsx

```diff
@@ -204,42 +226,49 @@ const Tooltip = ({
       // Adding true as third parameter to event listener will detect nested scrolls.
       window.addEventListener('scroll', onScrollDetected, true)
     }
-  }, [
-    visibleInDom,
-    positions.tooltipPosition,
-    generatePositions,
-    placement,
-    text,
-    onScrollDetected,
-  ])
+
+    return () => {
+      window.removeEventListener('scroll', onScrollDetected, true)
+    }
+  }, [generatePositions, onScrollDetected, visibleInDom])
+
+  /**
+   * If tooltip has `visible` prop it means the tooltip is manually controlled through this prop.
+   * In this cas we don't want to display tooltip on hover, but only when `visible` is true.
+   */
+  useEffect(() => {
+    if (isControlled) {
+      onPointerEvent(visible)()
+    }
+  }, [isControlled, onPointerEvent, visible])
 
   /**
    * Will render children conditionally if children is a function or not.
    */
   const renderChildren = useCallback(() => {
     if (typeof children === 'function') {
       return children({
-        onBlur: onMouseEvent(false),
-        onFocus: onMouseEvent(true),
-        onMouseEnter: onMouseEvent(true),
-        onMouseLeave: onMouseEvent(false),
+        onBlur: !isControlled ? onPointerEvent(false) : () => {},
```

Comment:
Let's extract all of these empty functions to a top-level `noop` function to avoid creating them every time

```suggestion
function noop() {}
...
        onBlur: !isControlled ? onPointerEvent(false) : noop,
```